Check/Codebase/PhpCodeSniffer.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * @file
+ * Contains \SiteAudit\Check\Codebase\PhpCodeSniffer.
+ */
+
+use Symfony\Component\Process\Process;
+/**
+ * Class SiteAuditCheckCodebasePhpCodeSniffer.
+ */
+class SiteAuditCheckCodebasePhpCodeSniffer extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('PHP Code Sniffer Violations');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Run phpcs on custom code.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {
+    if (isset($this->registry['phpcs_path'])) {
+      return dt('Cannot find phpcs in path. Make sure that phpcs is present in the PATH or run composer install inside the site_audit directory to install all the dependencies');
+    }
+    if (isset($this->registry['custom_code'])) {
+      return dt('No custom code path specified');
+    }
+    return dt('Cannot find coding standards inside ' . $this->registry['phpcs_standard']);
+  }
+
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('No PHP code_sniffer violations.');
+  }
+
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {
+    $ret_val = '';
+    if (drush_get_option('html') == TRUE) {
+      $ret_val .= '<table class="table table-condensed">';
+      $ret_val .= '<thead><tr><th>' . dt('Line, Column') . '</th><th>' . dt('Severity') . '</th><th>' . dt('Action') . '</th></tr></thead>';
+      foreach ($this->registry['phpcs_out'] as $filename => $violations) {
+        $num_violations = count($violations);
+        $ret_val .= "<tr align='center'><td colspan='3'><a href='file:///$filename'>File: $filename</a> Violations: $num_violations</td></tr>";
```

Comment:
Don't link to the files, please, strongest probability is that the report will be read in a place that doesn't have access to the files.